### PR TITLE
Tricks doc

### PR DESCRIPTION
**Pull request by DimaGrinevich**

### CODE REVIEW

File: docs/advanced/driver.md [hunk 1]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
...
+     //Page declared and initialized right before using it
+     HomePage homePage = new HomePage(getDriver());
+     homePage.open();
+     FooterMenu footerMenu = homePage.getFooterMenu();
+     CompareModelsPage comparePage = footerMenu.openComparePage();
+     List<ModelSpecs> specs = comparePage.compareModels("Samsung Galaxy J3", "Samsung Galaxy J5", "Samsung Galaxy J7 Pro");
+  }
+}
+```
+Not like this:
```
Comment (vdelendik) [inline docs/advanced/driver.md:133]: don't provide negative example via code. Just announce that getDriver() in BeforeSuite/BeforeClass calls are unwanted!

File: docs/automation/web.md [hunk 2]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+	// Verify model announced dates
+	SoftAssert softAssert = new SoftAssert();
+	softAssert.assertEquals(specs.get(0).readSpec(SpecType.ANNOUNCED), "2016, March 31");
+	softAssert.assertEquals(specs.get(1).readSpec(SpecType.ANNOUNCED), "2015, June 19");
+	softAssert.assertEquals(specs.get(2).readSpec(SpecType.ANNOUNCED), "2017, June");
+	softAssert.assertAll();
+    }
+}
+```
+If there is no way to do so, make sure you use @Test(dependsOnMethods=XXX) tag:
```
Comment (vdelendik) [inline docs/automation/web.md:341]: Change notice to:
Use `dependsOnMethods` only if it really required by test login.

File: docs/advanced/driver.md [hunk 3]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
 ##Restart
 * **restartDriver()** quit current driver and start new one with the same capabilities
```
Comment (avabishchevich) [inline docs/advanced/driver.md:115]: quit **the** current driver and start **a** new one

File: docs/advanced/driver.md [hunk 4]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
...
 * **restartDriver(boolean isSameDevice)** quit current driver and start new one on the same device using `uuid` capability. It is fully compatible with [MCloud](https://github.com/zebrunner/mcloud) farm.
```
Comment (avabishchevich) [inline docs/advanced/driver.md:116]: quit **the** current driver and start **a** new one

File: docs/advanced/driver.md [hunk 5]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
...
+###Tricks
+####Init pages and drivers in places where they are used, 
+Correct way:
```
Comment (avabishchevich) [inline docs/advanced/driver.md:119]: **The** correct way

File: docs/advanced/driver.md [hunk 6]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
...
+     //Page declared and initialized right before using it
+     HomePage homePage = new HomePage(getDriver());
+     homePage.open();
+     FooterMenu footerMenu = homePage.getFooterMenu();
+     CompareModelsPage comparePage = footerMenu.openComparePage();
+     List<ModelSpecs> specs = comparePage.compareModels("Samsung Galaxy J3", "Samsung Galaxy J5", "Samsung Galaxy J7 Pro");
+  }
+}
+```
+Unwanted approach:
```
Comment (avabishchevich) [inline docs/advanced/driver.md:133]: **An** unwanted approach

File: docs/automation/web.md [hunk 7]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
 ```
 >Dynamic elements loading. 
 **waitForJSToLoad()** method was introduced in AbstractPage class. It uses JS under the hood and helps to wait till all dynamic web elements on the page are getting loaded.
```
Comment (avabishchevich) [inline docs/automation/web.md:312]: It uses JS under the hood and helps to wait till all **the** dynamic web elements on the page **are loaded.**

File: docs/automation/web.md [hunk 8]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+
+###Tricks
+####Independent tests is the best way for the automation!
+Correct example:
```
Comment (avabishchevich) [inline docs/automation/web.md:316]: **A** correct example

File: docs/automation/web.md [hunk 9]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+```
+//Right way to write tests:
```
Comment (avabishchevich) [inline docs/automation/web.md:318]: **The** right way

File: docs/automation/web.md [hunk 10]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+	// Verify model announced dates
+	SoftAssert softAssert = new SoftAssert();
+	softAssert.assertEquals(specs.get(0).readSpec(SpecType.ANNOUNCED), "2016, March 31");
+	softAssert.assertEquals(specs.get(1).readSpec(SpecType.ANNOUNCED), "2015, June 19");
+	softAssert.assertEquals(specs.get(2).readSpec(SpecType.ANNOUNCED), "2017, June");
+	softAssert.assertAll();
+    }
+}
+```
+Use dependsOnMethods only if it really required by test login:
```
Comment (avabishchevich) [inline docs/automation/web.md:341]: if it **is** really required by test login:

File: docs/automation/web.md [hunk 11]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+        SoftAssert() softAssert = new SoftAssert();
+        softAssert.assertEquals(specs.get(0).readSpec(SpecType.ANNOUNCED), "2016, March 31");
+        softAssert.assertEquals(specs.get(1).readSpec(SpecType.ANNOUNCED), "2015, June 19");
+        softAssert.assertEquals(specs.get(2).readSpec(SpecType.ANNOUNCED), "2017, June");
+        softAssert.assertAll();
+    }
+}
+```
+#### Operate with web elements in page classes only!
+Correct way:
```
Comment (avabishchevich) [inline docs/automation/web.md:388]: **The** correct way

File: docs/automation/web.md [hunk 12]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+
+
+//In test class:
+public void someTest() {
+   WikipediaHomePage wikipediaHomePage = new WikipediaHomePage(getDriver());
+   wikipediaHomePage.open();
+   WikipediaLocalePage wikipediaLocalePage = wikipediaHomePage.goToWikipediaLocalePage(getDriver());
+}
+```
+Unwanted approach:
```
Comment (avabishchevich) [inline docs/automation/web.md:424]: **An** unwanted approach

File: docs/configuration.md [hunk 13]
```diff
@@ -424,3 +424,19 @@ test_run_rules=PRIORITY=>!!P1;;OWNER=>Josh&&!!Jake;;TAGS=>feature=web&&!!type=sm
...
 	</tr>
 </table>
+
+###Tricks
+#### Pass params through _config.properties, not in code.
+```
+Will work both:
+1) putting parameters in _config.properties :
+   selenium_url=http://localhost:4444/wd/hub
+2) passing them it right in test:
```
Comment (avabishchevich) [inline docs/configuration.md:434]: passing **them** right in **the** test (omit 'it')

File: docs/configuration.md [hunk 14]
```diff
@@ -424,3 +424,19 @@ test_run_rules=PRIORITY=>!!P1;;OWNER=>Josh&&!!Jake;;TAGS=>feature=web&&!!type=sm
...
+public void testCompareModels() {
+   R.CONFIG.put("selenium_host", "http://localhost:4444/wd/hub");
+   HomePage homePage = new HomePage(getDriver());
+   homePage.open();
+   ...
+}
+But recommended to use 1) variant for initialization of all parameters
```
Comment (avabishchevich) [inline docs/configuration.md:441]: **Nevertheless, it is** recommended to use **the 1st** variant for initialization of all **the** parameters.

File: docs/advanced/driver.md [hunk 15]
```diff
@@ -114,3 +114,40 @@ To disable driver quit strategy completely and cotrol drivers init/quit on your
 ##Restart
 * **restartDriver()** quit current driver and start new one with the same capabilities
 * **restartDriver(boolean isSameDevice)** quit current driver and start new one on the same device using `uuid` capability. It is fully compatible with [MCloud](https://github.com/zebrunner/mcloud) farm.
+###Tricks
+####Init pages and drivers in places where they are used,
```
Comment (avabishchevich) [inline docs/advanced/driver.md:118]: remove the comma?

File: docs/automation/web.md [hunk 16]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+        softAssert.assertEquals(specs.get(1).readSpec(SpecType.ANNOUNCED), "2015, June 19");
+        softAssert.assertEquals(specs.get(2).readSpec(SpecType.ANNOUNCED), "2017, June");
+        softAssert.assertAll();
+    }
+}
+```
+#### Operate with web elements in page classes only!
+Correct way:
+```
+//In page class:
```
Comment (avabishchevich) [inline docs/automation/web.md:390]: In **the** page class

File: docs/automation/web.md [hunk 17]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+   }
+   throw new RuntimeException("No language ref was found");
+}
+
+public void openLangList() {
+   langListBtn.clickIfPresent();
+}
+
+
+//In test class:
```
Comment (avabishchevich) [inline docs/automation/web.md:417]: In **the** test class

File: docs/automation/web.md [hunk 18]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+public void someTest() {
+   WikipediaHomePage wikipediaHomePage = new WikipediaHomePage(getDriver());
+   wikipediaHomePage.open();
+   WikipediaLocalePage wikipediaLocalePage = wikipediaHomePage.goToWikipediaLocalePage(getDriver());
+}
+```
+Unwanted approach:
+```
+//In page class:
```
Comment (avabishchevich) [inline docs/automation/web.md:426]: In **the** page class

File: docs/automation/web.md [hunk 19]
```diff
@@ -310,3 +310,144 @@ component.assertUIObjectNotPresent();   // equals to Assert.assertTrue(!componen
...
+    
+@FindBy(xpath = "//div[@id='js-lang-lists']//a")
+private List<ExtendedWebElement> langList;
+
+public List<ExtendedWebElement> getLangList(){
+   return langList;
+}
+
+
+//In test class:
```
Comment (avabishchevich) [inline docs/automation/web.md:438]: In **the** test class

**Pull request merged**